dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/PushPublishBundleGeneratorTest.java

```diff
@@ -597,5 +605,80 @@ public void Test_GenerateBundle_PageWithTemplateDeleted_PageSuccessfullyAddedToB
 
     }
 
+    /**
+     * Given Scenario: If a file asset has zero files and CONTENT_ALLOW_ZERO_LENGTH_FILES set to true,
+     *                  we should be able to generate the bundle without issues.
+     * ExpectedResult: The bundle should be generated without issues and the page added to the bundle.
+     */
+    @Test
+    public void Test_GenerateBundle_FileAssetZeroBytes_ContentAllowZeroLengthFilesSetToTrue_Success() throws Exception {
+        final boolean allowZeroLengthFilesDefault = Config.getBooleanProperty("CONTENT_ALLOW_ZERO_LENGTH_FILES", true);
+        try {
+            //Set Property to True
+            Config.setProperty("CONTENT_ALLOW_ZERO_LENGTH_FILES", true);
+            //Create Folder
+            final Folder folder = new FolderDataGen().nextPersisted();
+            // Create File with zero length
+            final File file = File.createTempFile("testing-file", ".vtl");
+            FileUtil.write(file, "");
+            // Create File Asset with that file
+            final Contentlet fileAssetShown = new FileAssetDataGen(folder, file).nextPersisted();
+            ContentletDataGen.publish(fileAssetShown);
+
+            //Create bundle
+            final Bundle bundle = createBundle("TestBundle" + System.currentTimeMillis(), false,
+                    "");
+            //Add assets to the bundle
+            PublisherAPI.getInstance()
+                    .saveBundleAssets(Arrays.asList(fileAssetShown.getIdentifier()), bundle.getId(),
+                            systemUser);
+            //Generate Bundle, will return several dependencySet with the assets that will be added to the bundle
+            final PushPublisherConfig listOfAssetsInBundle = generateBundle(bundle.getId(),
+                    Operation.PUBLISH);
+            Assert.assertNotNull(listOfAssetsInBundle);
+            Assert.assertTrue(
+                    listOfAssetsInBundle.getContentlets().contains(fileAssetShown.getIdentifier()));
+        } finally {
+            //Set Property to Default Value
+            Config.setProperty("CONTENT_ALLOW_ZERO_LENGTH_FILES", allowZeroLengthFilesDefault);
+        }
+
+    }
+
+    /**
+     * Given Scenario: If a file asset has zero bytes and CONTENT_ALLOW_ZERO_LENGTH_FILES set to false,
+     *                    when you try to generate a bundle with that file asset it fails.
+     * ExpectedResult: DotBundleException should be thrown when trying to generate the bundle with a file with zero bytes.
+     */
+    @Test (expected = DotBundleException.class)
+    public void Test_GenerateBundle_FileAssetZeroBytes_ContentAllowZeroLengthFilesSetToFalse_DotBundleException() throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5701517)